<font size="10">Video Game Sales Case Study</font>

<font size="5">Overview:</font>

Data Sourced: https://www.kaggle.com/datasets/gregorut/videogamesales

Today, I will discuss/analyze a dataset about the list of video games that have more than 100,000 copies sold. I picked this dataset because video games are one of my favorite hobbies, and I know that other people enjoy video games as a pasttime as well. Also, this dataset shows the ranks of these video games that have more than 100,000 copies sold, what console the game is for, the year it was released, genre, etc. It provides a lot of interesting information that I can analyze and try to make a correlation between copies sold and some other variable. Besides this, the analysis will be helpful to fans of video games and gaming copmanies to see the top video game sales, and what type of factors make a game successful in terms of sales.

In [1]:
import pandas as pd
import numpy as np
import plotly
import plotly.express as px
import plotly.graph_objects as go

# plotly.io is a low-level interface for interacting with figures/
# plotly.io.templates lists available plotly templates
# https://plotly.com/python-api-reference/plotly.io.html
import plotly.io as pio

pd.set_option('display.max_columns', 20)

In [2]:
import pandas as pd

url = 'https://raw.githubusercontent.com/themuid/Final-Case-Study-Project/261a95bd2679309790bb65751c4e1f2e466da792/vgsales.csv'
df_vgs = pd.read_csv(url)

# Dataset is now stored in a Pandas Dataframe

<font size="5">Questions to answer today:</font>



1.   What are the top 10 highest selling video games of all time?
2.   How do video game sales look like for each publisher?
3.   What is the most popular selling video game for each genre?
4.   What factor(s) result in the most video game sales for a game?
5.   Does year have any correlation with video game sale ranking?





<font size="5">Exploratory Data Analysis (EDA) </font>

In [3]:
df_vgs = pd.read_csv('https://raw.githubusercontent.com/themuid/Final-Case-Study-Project/261a95bd2679309790bb65751c4e1f2e466da792/vgsales.csv')
df_vgs.head()

Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

<font size="5">Explanation:</font>

*   From this code, we get a table visualzing out data of 5 rows x 11 columns
*   It is defaulted to 5 rows as we have df_vgs.head() without a number in the parantheses.
*   If we want more rows, we can put a specific number in the parantheses.
*   Ex: df_.head(20) = 20 rows displayed

For our data set, every column is a relevant variable to analyzing our video game sales ranking, so no need to run any code to remove them. We can start by using the df.info() function to check the data of each column, and see the number of rows with non-missing values.

In [6]:
df_vgs.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


Now, we can also find the number of columns and rows of the Video Games Sales data by using df_shape and print functions.

In [4]:
num_rows, num_cols = df_vgs.shape
print(f'There are {num_rows} rows and {num_cols} columns in the dataset.')

There are 16598 rows and 11 columns in the dataset.


16598 rows x 11 columns

Now, we will aim to use (EDA) to answer our 1st question of the Case Study.

"What are the top 10 highest selling video games of all time?"